src/providers/ogr/qgsogrdataitems.cpp

```diff
@@ -110,6 +114,38 @@ QString QgsOgrLayerItem::layerName() const
     return info.completeBaseName();
 }
 
+#ifdef HAVE_GUI
+QList<QAction *> QgsOgrLayerItem::actions()
+{
+  QList<QAction *> lst;
+  QAction *actionDeleteLayer = new QAction( tr( "Delete %1" ).arg( mName ), this );
+  connect( actionDeleteLayer, &QAction::triggered, this, &QgsOgrLayerItem::deleteLayer );
+  lst.append( actionDeleteLayer );
+  return lst;
+}
+
+void QgsOgrLayerItem::deleteLayer()
+{
+  if ( QMessageBox::question( nullptr, QObject::tr( "Delete Layer" ),
+                              QObject::tr( "Are you sure you want to delete %1?" ).arg( mName ),
```

Comment:
Not sure: it might be a table in a DB and not a file. 
But, I was thinking at changing into: `"Are you sure you want to delete '%1'?..."` , do we have a conventions that when an action opens a dialog we place the ellipsis? We have a confirmation dialog opening here.